### PR TITLE
feature: fuzzing 'Connect-As' field

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -363,11 +363,11 @@ The test index makes use of symbolic language in describing connection and messa
 
 ### ZG-RESISTANCE-003
 
-    The node rejects the handshake when there is a bit flip in either a public_key or a shared_value (which later results in
-    an invalid session signature). This should happen for both types of connection: Initiator and Responder.
-    These tests attempt handshakes with intentionally broken public_key and signature and assert whether
-    rippled dropped connection after some short time.
-
+    The node rejects the handshake when there is an in any of handshake field: 
+    1. Bit flip in either a public_key.
```
Comment (kylegranger) [inline SPEC.md:367]: `either` could be cut here

File: src/tests/resistance/handshake.rs [hunk 2]
```diff
@@ -242,6 +242,54 @@ async fn r001_t3_HANDSHAKE_connection_field() {
...
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+    let cfg = gen_cfg("PeerPeer".to_owned());
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Find the largest instance value which the node could accept, but won't due to invalid value
+    // in the field.
+    let cfg = gen_cfg(gen_huge_string(WS_HTTP_HEADER_MAX_SIZE));
+    assert!(!run_handshake_req_test_with_cfg(cfg, debug).await);
+
+    // Use a huge value which the node will always reject.
```
Comment (kylegranger) [inline src/tests/resistance/handshake.rs:284]: minor grammar fix: `which the node...` s.b. `that the node`
(`which` is subject of verb phrase)

**Review by kylegranger — APPROVED** [review]
👍

File: SPEC.md [hunk 3]
```diff
@@ -363,11 +363,11 @@ The test index makes use of symbolic language in describing connection and messa
...
-    The node rejects the handshake when there is a bit flip in either a public_key or a shared_value (which later results in
-    an invalid session signature). This should happen for both types of connection: Initiator and Responder.
-    These tests attempt handshakes with intentionally broken public_key and signature and assert whether
-    rippled dropped connection after some short time.
-
+    The node rejects the handshake when there is an in any of handshake field: 
+    1. Bit flip in either a public_key. 
+    2. Bit flip in a shared_value (which later results in an invalid session signature). 
+    3. "Connection" field is incorrect.
+    4. "Connect-As" field is incorrect.
```
Comment (Rqnsom) [inline SPEC.md:370]:
```suggestion
    3. "Connection" HTTP field is incorrect.
    4. "Connect-As" HTTP field is incorrect.
```

**Review by Rqnsom — APPROVED** [review]
LGTM 🚀

File: SPEC.md (hunk 1 above)
Comment (lwojtow) [reply SPEC.md]: Done

File: src/tests/resistance/handshake.rs (hunk 2 above)
Comment (lwojtow) [reply src/tests/resistance/handshake.rs]: Thanks, done.

File: SPEC.md (hunk 3 above)
Comment (lwojtow) [reply SPEC.md]: Done

**Pull request merged**